cardano-cli/src/Cardano/CLI/EraBased/Options/Genesis.hs

```diff
@@ -211,7 +211,7 @@ pGenesisCreateTestNetData envCli =
     <*> pNumUtxoKeys
     <*> pSupply
     <*> pSupplyDelegated
-    <*> pNetworkId envCli
+    <*> (optional $ pNetworkId envCli)
```

Comment:
You should update the parser's help text so the user knows this value will override the Genesis file's value.

Reply:
@CarlosLopezDeLara> when you do so, you will need to regenerate the golden files as follows:

```
RECREATE_GOLDEN_FILES=1 cabal test all
```